lua/core/menu/devices.lua

```diff
@@ -110,26 +130,55 @@ m.enc = function(n,delta)
   end
 end
 
+local function redraw_keyboard_layout_select()
+  local len = tabutil.count(m.options["keyboard layout"])
+  screen.clear()
+  for i=1,6 do
+    if (i > 2 - m.pos + 1) and (i < m.len - m.pos + 3 + 1) then
+      local name = m.options["keyboard layout"][i+m.pos-2 - 1]
+      local line = string.upper(name)
+      screen.level(i==3 and 15 or 4)
+      screen.move(20,10*i)
+      screen.text(line)
+    end
+  end
+  screen.update()
+end
+
 m.redraw = function()
+
+  if (m.section == "keyboard layout" and m.mode == "select") then
+    redraw_keyboard_layout_select()
+    return
+  end
+
   local y_offset = 0
-  if(4<m.pos) and not (m.section == "midi" and m.mode == "list") then
+  if not (m.mode == "type")
+    and not (m.section == "midi" and m.mode == "list")
+    and (4<m.pos) then
     y_offset = 10*(4-m.pos)
+  elseif m.mode ~= "type" then
+    y_offset = 20
```

Comment:
incorporated the systematic `+20` as part of the `y_offset` and disabled it case  `m.mode = "type"`